contentgrid-spring-boot-autoconfigure/src/main/java/com/contentgrid/spring/boot/autoconfigure/integration/EventsAutoConfiguration.java

```diff
@@ -0,0 +1,92 @@
+import com.contentgrid.spring.integration.events.ContentGridHalAssembler;
+import com.contentgrid.spring.integration.events.ContentGridMessageHandler;
+import com.contentgrid.spring.integration.events.ContentGridPublisherEventListener;
+import com.contentgrid.spring.integration.events.EntityToPersistentEntityResourceTransformer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@Configuration
+@ConditionalOnClass(ContentGridPublisherEventListener.class)
+@ConditionalOnBean(TypeConstrainedMappingJackson2HttpMessageConverter.class)
+@AutoConfigureAfter(RepositoryRestMvcAutoConfiguration.class)
+@IntegrationComponentScan(basePackageClasses = ContentGridEventPublisher.class)
+@EnableConfigurationProperties(EventConfigurationProperties.class)
+public class EventsAutoConfiguration {
+
+    @Bean
+    IntegrationFlow toOutboundQueueFlow(EventConfigurationProperties config,
+            ObjectProvider<ContentGridMessageHandler> handlers,
+            @Qualifier("halJacksonHttpMessageConverter") TypeConstrainedMappingJackson2HttpMessageConverter typeConstrainedMappingJackson2HttpMessageConverter,
+            ApplicationContext context) {
+
+        ObjectMapper halObjectMapper = typeConstrainedMappingJackson2HttpMessageConverter
+                .getObjectMapper();
+
+        IntegrationFlowBuilder builder = IntegrationFlows
+                .from(ContentGridEventPublisher.CONTENTGRID_EVENT_CHANNEL)
+                .transform(new EntityToPersistentEntityResourceTransformer(
+                        new ContentGridHalAssembler(context)))
+                .transform(Transformers.toJson(new Jackson2JsonObjectMapper(halObjectMapper),
+                        MediaTypes.HAL_JSON_VALUE));
+
+        handlers.stream().map(ContentGridMessageHandler::get).forEach(builder::handle);
+        return builder.get();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    // TODO we can exclude this autoconfig by configuration property
+    ContentGridPublisherEventListener contentGridPublisherEventListener(
+            ContentGridEventPublisher contentGridEventPublisher,
+            EntityManagerFactory entityManagerFactory) {
+        return new ContentGridPublisherEventListener(contentGridEventPublisher,
+                entityManagerFactory);
+    }
+
+    @Configuration
+    @ConditionalOnProperty(value = { "spring.rabbitmq.host" })
+    static class EventsRabbitMqAutoConfiguration {
+
+        @Bean
+        @ConditionalOnProperty(value = { "contentgrid.events.queueName" })
+        Queue queue(EventConfigurationProperties config) {
+            return new Queue(config.getQueueName(), false);
+        }
+
+        @Bean
+        // TODO do we need a conditional here?
+        ContentGridMessageHandler messageHandler(RabbitTemplate rabbitTemplate,
+                EventConfigurationProperties config) {
+            return () -> Amqp.outboundAdapter(rabbitTemplate).routingKey(config.getQueueName());
```

Comment:
so technically `EventConfigurationProperties` should not have a `queue-name` property, but a `routing-key` property instead ?

Reply:
fair point - but can we use `contentgrid.events.queue-name` instead :-)

Reply:
`contentgrid.events.rabbitmq.routing-key`